lib/msal-angular/src/msal-guard.service.ts

```diff
@@ -1,96 +1,60 @@
+import { MSAL_CONFIG, MsalService, MSAL_CONFIG_ANGULAR } from "./msal.service";
+import { Location, PlatformLocation } from "@angular/common";
+import { BroadcastService } from "./broadcast.service";
+import { Configuration, AuthResponse, AuthError } from "msal";
+import { MsalAngularConfiguration } from "./msal-angular.configuration";
 
 @Injectable()
 export class MsalGuard implements CanActivate {
 
-    constructor(@Inject(MSAL_CONFIG) private config: MsalConfig, private authService: MsalService, private router: Router, private activatedRoute: ActivatedRoute, private location: Location, private platformLocation: PlatformLocation, private broadcastService: BroadcastService) {
-    }
+    constructor(
+        @Inject(MSAL_CONFIG) private msalConfig: Configuration,
+        @Inject(MSAL_CONFIG_ANGULAR) private msalAngularConfig: MsalAngularConfiguration,
+        private authService: MsalService,
+        private router: Router,
+        private activatedRoute: ActivatedRoute,
+        private location: Location,
+        private platformLocation: PlatformLocation,
+        private broadcastService: BroadcastService
+    ) {}
 
     canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean | Promise<boolean> {
         this.authService.getLogger().verbose("location change event from old url to new url");
 
-        this.authService.updateDataFromCache([this.config.clientID]);
-        if (!this.authService._oauthData.isAuthenticated && this.isObjectEmpty(this.authService._oauthData.idToken)) {
-            if (state.url) {
-
-                if (!this.authService._renewActive && !this.authService.loginInProgress()) {
-
-                    var loginStartPage = this.getBaseUrl() + state.url;
-                    if (loginStartPage !== null) {
-                        this.authService.getCacheStorage().setItem(Constants.angularLoginRequest, loginStartPage);
-                    }
-                    if (this.config.popUp) {
-                        return this.authService.loginPopup(this.config.consentScopes, this.config.extraQueryParameters)
-                            .then(function (token) {
-                               return true;
-                            }, function (error) {
-                                return false;
-                            });
-                    }
-                    else {
-                        this.authService.loginRedirect(this.config.consentScopes, this.config.extraQueryParameters);
-                    }
-                }
+        if (!this.authService.getAccount()) {
+            if (this.msalAngularConfig.popUp) {
+                return this.authService.loginPopup({
+                    scopes: this.msalAngularConfig.consentScopes,
+                    extraQueryParameters: this.msalAngularConfig.extraQueryParameters
+                })
+                    .then(() => true)
+                    .catch(() => false);
```

Comment:
more questions about guards i guess... do guards not inherently have something like this implemented in them? from what we talked about earlier, it seemed like this was a part of their purpose and may be built in.

if that isn't the case, this is fine, just curious

Reply:
It's on you to implement the desired business logic for your guard, but ultimately, you need to return a boolean indicating whether or not the user has met the criteria for accessing the desired page.